timemachine/cpp/src/device_buffer.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include <cstdlib>
+
+namespace timemachine {
+
+template <typename T> class DeviceBuffer {
+
+public:
+    DeviceBuffer(const std::size_t length);
+
+    ~DeviceBuffer();
+
+    const size_t size;
```

Comment:
I would probably make this a bit more C++ safe, right now the caller can just directly set this. You probably want to move this to a private member, and *only* allow for a `size()` function that returns the size, but disable the setter (by simply not implementing one. I can't think of a good use case for why we'd want to the setter

Reply:
Actually sorry I confused, I didn't realize this was `const` - which means it can only be set in the constructor